### PR TITLE
fix(monitoring): pass explicit batch_size to self.log in LightningModules

### LINKED ISSUE #600: fix(monitoring): pass explicit batch_size= to self.log() in LightningModules
## Summary

Lightning emits a warning when it can't auto-detect batch size from a non-tensor batch (dicts, tuples, lists). Our batches aren't simple tensors, so every training run prints a noisy `Trying to infer the `batch_size` from an ambiguous collection` warning with the inferred value (e.g. `1024`). The value is correct; the warning is harmless; but it's trivially fixable by passing `batch_size=` explicitly to each `self.log(...)` call.

## Background

Lightning uses `batch_size` to correctly average metrics across steps/epochs. When it can't infer it confidently, it picks a value and warns. Passing it explicitly silences the warning and protects future runs from structure changes.

## Affected files

The following modules contain `self.log(...)` calls without a `batch_size=` argument:

- `src/models/ksin_flow_matching_module.py`
- `src/models/surge_flowvae_module.py`
- `src/models/ksin_ff_module.py`

(There may be others — grep the full `src/models/` tree to catch them all.)

## Fix

Pass `batch_size=<batch_size>` explicitly to each `self.log(...)` call, where `<batch_size>` is derived from the input tensor's first dimension (typically `x.shape[0]` or equivalent for the specific module).

Example:

```python
# before
self.log("train/loss", loss, on_step=True, on_epoch=True, prog_bar=True)

# after
self.log("train/loss", loss, on_step=True, on_epoch=True, prog_bar=True, batch_size=x.shape[0])
```

Double-check which tensor in each module's `training_step` / `validation_step` / `test_step` is the correct source for batch size — it may vary (e.g., `audio.shape[0]`, `params.shape[0]`).

## Acceptance criteria

- [ ] No `Trying to infer the batch_size` warning appears during `make test` or a standard `python src/train.py experiment=kosc/ffn_mse` run.
- [ ] All `self.log(...)` calls in the affected modules (and any others found via grep) have explicit `batch_size=`.
- [ ] Existing tests still pass.

## Good First Issue

This is a small, well-scoped, mechanical change — a good entry point for new contributors. No architectural decisions required.

Part of #107.

### PR DESCRIPTION
**Pull request by Devisri-B**
## Why

PyTorch Lightning prints a `PossibleUserWarning` on every training run: "Trying to infer the `batch_size` from an ambiguous collection". The LightningModules call `self.log` / `self.log_dict` without an explicit `batch_size`, and the batch is a non-tensor collection (a tuple for the k-sinusoidal models, a dict for the Surge Flow-VAE), so Lightning cannot infer the size unambiguously. The guessed value is correct, but the warning is noise and leaves epoch-level reductions relying on a guess.

Closes #600

## What changed

- Source the batch size from the input tensor's leading dimension and pass `batch_size=` to every `self.log` / `self.log_dict` call in the `training_step` / `validation_step` / `test_step` of `ksin_ff_module`, `ksin_flow_matching_module`, and `surge_flowvae_module`.
- `surge_flowvae` unpacks `mel_spec` from `model_step` instead of re-indexing the batch, keeping the tuple-typed step signatures type-clean.
- `ksin_flow_matching` logs grad norms in `on_before_optimizer_step` (`on_epoch=True`), which has no batch; it reuses a batch size cached in `training_step` (`self._batch_size`).
- Added a unit test (every `self.log` in the ksin feed-forward steps passes `batch_size`) and an integration test (a `fast_dev_run` run emits no ambiguous-`batch_size` warning, captured directly since the suite ignores `UserWarning`).
- Also guard `KSinFlowMatchingModule.training_step` against a `None` penalty (`return loss + penalty` → `return loss if penalty is None else loss + penalty`), a latent `TypeError` flagged in review.

## Test plan

- Re-ran the reproduction `pytest tests/test_train.py::test_train_fast_dev_run_tiny_model_tiny_data -W always`: the `ambiguous collection` warning printed before this change and is gone after.
- New `test_train_fast_dev_run_emits_no_batch_size_warning` passes.
- `make test-fast` is green locally except `test_pinned_baselines_resolve`, a `@pytest.mark.network` baseline-fetch test that can't reach the network in my local environment (unrelated to this change).

## Out of scope

- The other Surge modules (`surge_ff_module.py`, `surge_flow_matching_module.py`, `surge_fake_oracle_module.py`) share the pattern but are outside this issue's named files — happy to follow up separately.

______________________________________________________________________

- [x] One logical change; any breaking changes are called out above.

### CODE REVIEW

File: src/synth_setter/models/ksin_flow_matching_module.py [hunk 1]
```diff
@@ -369,10 +373,21 @@ def training_step(self, batch: tuple[torch.Tensor, torch.Tensor], batch_idx: int
...
+            self.log(
+                "train/penalty",
+                penalty,
+                on_step=True,
+                on_epoch=True,
+                prog_bar=True,
+                batch_size=batch_size,
+            )
 
         return loss + penalty
```
[the remark this answers is not in the document]
Comment (Devisri-B) [reply src/synth_setter/models/ksin_flow_matching_module.py]: Good catch on the latent None case. This is pre-existing `return loss + penalty` is
unchanged by this PR (which only adds batch_size= to the log calls) and is unrelated to
#600 so I've left it out of scope here. Happy to open a separate issue/PR to guard it
(e.g. `return loss if penalty is None else loss + penalty`).

File: src/synth_setter/models/surge_flowvae_module.py [hunk 2]
```diff
@@ -68,42 +68,53 @@ def get_beta(self) -> float:
         )
 
     def training_step(self, batch: tuple[torch.Tensor, torch.Tensor], batch_idx: int):
-        losses, *_, vae_out = self.model_step(batch)
+        losses, mel_spec, _, vae_out = self.model_step(batch)
```
[the remark this answers is not in the document]
Comment (Devisri-B) [reply src/synth_setter/models/surge_flowvae_module.py]: Done, changed the batch annotation on training_step, validation_step, test_step (and predict_step, which had the same mismatch) to dict[str, torch.Tensor], matching model_step and the actual batch. pyright is clean. Fixed in 4394fc0.

File: src/synth_setter/models/surge_flowvae_module.py [hunk 3]
```diff
@@ -68,42 +68,53 @@ def get_beta(self) -> float:
...
+        self.log("train/beta", beta, on_step=True, prog_bar=True, batch_size=batch_size)
 
         return loss
 
     def on_train_epoch_end(self) -> None:
         pass
 
     def validation_step(self, batch: tuple[torch.Tensor, torch.Tensor], batch_idx: int):
-        losses, *_, vae_out = self.model_step(batch)
+        losses, mel_spec, _, vae_out = self.model_step(batch)
```
[the remark this answers is not in the document]
Comment (Devisri-B) [reply src/synth_setter/models/surge_flowvae_module.py]: Done, changed the batch annotation on training_step, validation_step, test_step (and predict_step, which had the same mismatch) to dict[str, torch.Tensor], matching model_step and the actual batch. pyright is clean. Fixed in 4394fc0.

File: src/synth_setter/models/surge_flowvae_module.py [hunk 4]
```diff
@@ -68,42 +68,53 @@ def get_beta(self) -> float:
...
         losses = {f"val/{k}": v for k, v in losses.items()}
-        self.log_dict(losses, on_step=False, on_epoch=True, prog_bar=True)
+        self.log_dict(losses, on_step=False, on_epoch=True, prog_bar=True, batch_size=batch_size)
 
     def on_validation_epoch_end(self):
         pass
 
     def test_step(self, batch: tuple[torch.Tensor, torch.Tensor], batch_idx: int):
-        losses, *_ = self.model_step(batch)
+        losses, mel_spec, *_ = self.model_step(batch)
```
[the remark this answers is not in the document]
Comment (Devisri-B) [reply src/synth_setter/models/surge_flowvae_module.py]: Done, changed the batch annotation on training_step, validation_step, test_step (and predict_step, which had the same mismatch) to dict[str, torch.Tensor], matching model_step and the actual batch. pyright is clean. Fixed in 4394fc0.

File: tests/test_train.py [hunk 5]
```diff
@@ -78,6 +79,30 @@ def test_train_fast_dev_run_tiny_model_tiny_data(cfg_train: DictConfig) -> None:
...
+    """
+    HydraConfig().set_config(cfg_train)
+    with open_dict(cfg_train):
+        cfg_train.trainer.fast_dev_run = True
+
+    with warnings.catch_warnings(record=True) as caught:
+        warnings.simplefilter("always")
+        train(cfg_train)
+
+    batch_size_warnings = [str(w.message) for w in caught if "batch_size" in str(w.message)]
```
[the remark this answers is not in the document]
Comment (Devisri-B) [reply tests/test_train.py]: Good call, tightened this to match Lightning's specific message instead of any "batch_size" substring. Done in 2ac6998.

**Comment by Devisri-B** [comment]
Hi @ktinubu, first contribution here. This addresses #600 by passing an explicit batch_size to the self.log calls across the three LightningModules, silencing the "ambiguous collection" warning on every training run. Added a unit test and an integration test. Would appreciate a review when you have time. Thanks!

File: src/synth_setter/models/ksin_flow_matching_module.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Devisri-B) [reply src/synth_setter/models/ksin_flow_matching_module.py]: On reflection, since it's a one-line change I went ahead and applied your suggested fix: `return loss if penalty is None else loss + penalty`. Fixed in 5189ac9. Thanks for catching it!

**Comment by ktinubu** [comment]
Closing after stale-PR triage: most target modules were removed or restructured. Issue #600 remains valid, but it needs a fresh patch against the current VSTFlowVAEModule.
